### PR TITLE
Update echo to version 3

### PR DESCRIPTION
**Pull request by joelnb**
I believe this pull request does everything required to enable the use of the latest version (v3) of github.com/labstack/echo which has an API different to the current version in use. It is inspired by the pull request from @rtuin and built on top of the work by @phuongvu to update to v2.

Currently I have added a version constraint on `^3.0.0` for echo to the `glide.yaml` (to show that it is not compatible with earlier versions and to hopefully enable easier use of `glide update` in future) but I can remove that if you would prefer to just rely on the `glide.lock` for controlling the version of dependencies.

This pull request also update the version of all other dependencies as this was required to use the new version of echo (hence the massive number of changes).

### CODE REVIEW

**Comment by olebedev** [comment]
@joelnb, it's better to rely on the glide.lock.

**Comment by joelnb** [comment]
Okay, I pushed another commit which removes that from the `glide.yaml` & updates the `glide.lock`.

File: server/react.go [hunk 1]
```diff
@@ -65,8 +63,8 @@ func (r *React) Handle(c echo.Context) error {
 	start := time.Now()
 	select {
 	case re := <-vm.Handle(map[string]interface{}{
-		"url":     c.Request().(*standard.Request).URI(),
-		"headers": c.Request().(*standard.Request).Request.Header,
+		"url":     c.Request().URL.Path,
```
Comment (olebedev) [inline server/react.go]: This should be a `URL.String()` instead of `URL.Path`.

**Comment by joelnb** [comment]
@olebedev thanks for the review. I corrected that error and also added 2 more commits which remove the gzip middleware (as you suggested in https://github.com/olebedev/go-starter-kit/pull/55) and revert the `e` variable to being called `engine` as it seems a more descriptive name (and is the name currently in master anyway so reduces the changes made here).

**Comment by olebedev** [comment]
Thanks!

**Comment by rtuin** [comment]
Very nice, thanks!

**Comment by olebedev** [comment]
@rtuin could you leave some thoughts here #68 ?

**Pull request merged**